src/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -23,6 +24,13 @@ public ObjectCollection(Action<T> validator)
             _validator = validator;
         }
 
+        // This is only used internally to enumerate the collection
+        // without the enumerator allocation.
+        new public List<T>.Enumerator GetEnumerator()
```

Comment:
It's unfortunate that `Collection<T>` (and `ReadOnlyCollection<T>`) don't have a struct-based enumerator, but I don't think there's anyway to compatibly add one at this point because `GetEnumerator()` is already public and returns `IEnumerator<T>` :(